dosbox-staging/dosbox-staging
# Unicode API refactoring

# Description

- Unicode routines now have their own header file - `string_utils.h`, IMHO, grew too crowded
- DOS<->Unicode conversion API was reworked; there is no `bool` return value anymore (it's usage in current code was replaced by searching the output string for `\0` characters)
- some minor internal changes


# Manual testing

- checked that `intro` command still works, even with `language = pl` setting
- checked that `tree` command still works, even after `keyb br 3848` command (in this case it should use fallback ASCII artwork)
- using country test program available in other PR (https://github.com/dosbox-staging/dosbox-staging/pull/3164) I have checked that for `country = 48` (Poland) currency symbol is `zł` with code page 668 (execute _keyb pl 668_ command) and `zl` (no stroke on `l`) for code page 437;
- using the same tool I have checked that for `country = 49` and `locale_period = modern`  currency symbol is `EUR` with code page 437 (`keyb us 437` command) and `€` with code page 858 (`keyb us 858` command)

# Checklist

I have:
